ts/Series/TiledWebMap/TiledWebMapSeries.ts

```diff
@@ -0,0 +1,698 @@
+        return { x: xTile, y: yTile };
+    }
+
+    /**
+     * Convert tile to map coordinates in longitude/latitude
+     *
+     * @function Highcharts.MapView#tileToLonLat
+     * @since  next
+     * @param  xTile
+     *         Position x of the tile
+     * @param  yTile
+     *         Position y of the tile
+     * @param  zTile
+     *         Zoom of the tile
+     * @return {Highcharts.MapLonLatObject}
+     *         The map coordinates
+     */
+
+    public tileToLonLat(
+        xTile: number,
+        yTile: number,
+        zTile: number
+    ): Highcharts.MapLonLatObject {
+        const lon = xTile / Math.pow(2, zTile) * 360 - 180,
+            n = Math.PI - 2 * Math.PI * yTile / Math.pow(2, zTile),
+            lat = (
+                180 /
+                Math.PI * Math.atan(0.5 * (Math.exp(n) - Math.exp(-n)))
+            );
+        return { lon, lat };
+    }
+
+    public drawPoints(): void {
+        const {
+                chart
+            } = this,
+            mapView = chart.mapView;
+
+        if (!this.tiles) {
+            this.tiles = {};
+        }
+        if (!this.transformGroups) {
+            this.transformGroups = [];
+        }
+        if (!mapView) {
+            return;
+        }
+
+        const {
+                tiles,
+                transformGroups
+            } = this,
+            options = this.options,
+            provider = options.provider,
+            { zoom } = mapView,
+            worldSize = 400.979322,
+            tileSize = 256;
+        let zoomFloor = zoom < 0 ? 0 : Math.floor(zoom),
+            maxTile = Math.pow(2, zoomFloor),
+            scale = ((tileSize / worldSize) * Math.pow(2, zoom)) /
+                ((tileSize / worldSize) * Math.pow(2, zoomFloor)),
+            scaledTileSize = scale * 256;
+
+        if (provider && (provider.type || provider.url)) {
+            if (provider.type && !provider.url) {
+                const ProviderDefinition =
+                    TilesProvidersRegistry[provider.type];
+
+                if (!defined(ProviderDefinition)) {
+                    error(
+                        'Provider cannot be reached.',
+                        false
+                    );
+                    return;
+                }
+
+                const def = new ProviderDefinition(),
+                    { initialProjectionName: providerProjection } = def;
+                let theme,
+                    subdomain = '';
+
+                if (provider.theme && defined(def.themes[provider.theme])) {
+                    theme = def.themes[provider.theme];
+                } else {
+                    // if nothing set take first theme
+                    theme = def.themes[Object.keys(def.themes)[0]];
+                    error(
+                        'Missing option: Tiles provider theme cannot be ' +
+                        'reached, using standard provider theme.',
+                        false
+                    );
```

Comment:
Do we want to stick with the current generic error messages, or we want to be more specific about them? 
Example:
User sets provider theme to "foo"
Do we want to display a message that contains it? Like "Tiles provider "foo" theme cannot be reached..."

Reply:
@KacperMadej can you share your thoughts about this?

Reply:
If the reason for the failure is clear then the message should also be clear. Please add the detailed reason for the error if we are sure why it failed. When using defaults - no need to throw for that. A better message would be to throw a warning when the set theme was not defined and it fell down to the default - something like "The Provider's Theme '${provider.theme}' is not defined in the Provider definition - falling back to '${theme}'." That way at least something works but looks not as expected and checking the console will give a helpful message. The same could be done for subdomain.

Reply:
Ok, so we will stick with warnings for using defaults.